.github/workflows/ci.yaml

```diff
@@ -29,6 +29,17 @@ jobs:
         run: yarn test
       - name: Integration Test
         run: yarn test:integration
+  test-build:
+    needs: [test]
+    runs-on: ubuntu-18.04
+    if: github.ref != 'refs/heads/master'
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build test
+        uses: docker/build-push-action@v4
+        with:
+          load: true
+          tags: ghcr.io/elifesciences/enhanced-preprints:test
```

Comment:
Ideally we would add:

```
docker run ghcr.io/elifesciences/enhanced-preprints:test -d
```

Then check the `localhost:3000/api/reviewed-preprints` returned a status of `200`. I tried to implement this but failed. Maybe could pair on this tomorrow.

I have verified that this job would have caused the 2 recent renovate PR's that caused builds to fail would have failed ahead of merge.

See:
- https://github.com/elifesciences/enhanced-preprints-server/pull/600
- https://github.com/elifesciences/enhanced-preprints-server/pull/601

The commit that proves CI would have failed is: https://github.com/elifesciences/enhanced-preprints-server/actions/runs/4317177023/jobs/7533933462

Reply:
> Ideally we would add:
> 
> ```
> docker run ghcr.io/elifesciences/enhanced-preprints:test -d
> ```
> 
> Then check the `localhost:3000/api/reviewed-preprints` returned a status of `200`

It's probably due to EPP server taking a little time to start up and bind to the socket. This patch works locally in act:
```
> git diff
diff --git a/.github/workflows/ci.yaml b/.github/workflows/ci.yaml
index df445bd..5071777 100644
--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -43,7 +43,7 @@ jobs:
         run: docker run -d --name myapp -p 3000:3000 ghcr.io/elifesciences/enhanced-preprints:test
       - name: Check status code
         run: |
-          curl --verbose http://localhost:3000/api/reviewed-preprints
+          curl --retry 5 --retry-max-time 10 --retry-connrefused --verbose http://localhost:3000/api/reviewed-preprints
   build-and-push:
     needs: [test]
     runs-on: ubuntu-18.04

> act -j test-build
WARN  ⚠ You are using Apple M1 chip and you have not specified container architecture, you might encounter issues while running act. If so, try running it with '--container-architecture linux/amd64'. ⚠  
[Pipeline/test-build] 🚀  Start image=ghcr.io/catthehacker/ubuntu:act-18.04
[Pipeline/test-build]   🐳  docker pull image=ghcr.io/catthehacker/ubuntu:act-18.04 platform= username= forcePull=true
[Pipeline/test-build]   🐳  docker create image=ghcr.io/catthehacker/ubuntu:act-18.04 platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
[Pipeline/test-build]   🐳  docker run image=ghcr.io/catthehacker/ubuntu:act-18.04 platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
[Pipeline/test-build]   ☁  git clone 'https://github.com/docker/build-push-action' # ref=v4
[Pipeline/test-build] ⭐ Run Main actions/checkout@v3
[Pipeline/test-build]   🐳  docker cp src=/Users/scottaubrey/Sources/elife/enhanced-preprints-server/. dst=/Users/scottaubrey/Sources/elife/enhanced-preprints-server
[Pipeline/test-build]   ✅  Success - Main actions/checkout@v3
[Pipeline/test-build] ⭐ Run Main Build test
[Pipeline/test-build]   🐳  docker cp src=/Users/scottaubrey/.cache/act/docker-build-push-action@v4/ dst=/var/run/act/actions/docker-build-push-action@v4/
[Pipeline/test-build]   🐳  docker exec cmd=[node /var/run/act/actions/docker-build-push-action@v4/dist/index.js] user= workdir=
[Pipeline/test-build]   ❓  ::group::GitHub Actions runtime token access controls
| ACTIONS_RUNTIME_TOKEN not set
[Pipeline/test-build]   ❓  ::endgroup::
[Pipeline/test-build]   ❓  ::group::Docker info
| [command]/usr/bin/docker version
| Client:
|  Version:           20.10.22+azure-1
|  API version:       1.41
|  Go version:        go1.18.9
|  Git commit:        3a2c30b63ab20acfcc3f3550ea756a0561655a77
|  Built:             Thu Dec 15 15:37:38 UTC 2022
|  OS/Arch:           linux/arm64
|  Context:           default
|  Experimental:      true
| 
| Server: Docker Desktop 4.17.0 (99724)
|  Engine:
|   Version:          20.10.23
|   API version:      1.41 (minimum version 1.12)
|   Go version:       go1.18.10
|   Git commit:       6051f14
|   Built:            Thu Jan 19 17:31:28 2023
|   OS/Arch:          linux/arm64
|   Experimental:     false
|  containerd:
|   Version:          1.6.18
|   GitCommit:        2456e983eb9e37e47538f59ea18f2043c9a73640
|  runc:
|   Version:          1.1.4
|   GitCommit:        v1.1.4-0-g5fd4c4d
|  docker-init:
|   Version:          0.19.0
|   GitCommit:        de40ad0
| [command]/usr/bin/docker info
| Client:
|  Context:    default
|  Debug Mode: false
|  Plugins:
|   buildx: Docker Buildx (Docker Inc., 0.10.0+azure-1)
|   compose: Docker Compose (Docker Inc., 2.15.1+azure-1)
| 
| Server:
|  Containers: 5
|   Running: 2
|   Paused: 0
|   Stopped: 3
|  Images: 16
|  Server Version: 20.10.23
|  Storage Driver: overlay2
|   Backing Filesystem: extfs
|   Supports d_type: true
|   Native Overlay Diff: true
|   userxattr: false
|  Logging Driver: json-file
|  Cgroup Driver: cgroupfs
|  Cgroup Version: 2
|  Plugins:
|   Volume: local
|   Network: bridge host ipvlan macvlan null overlay
|   Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
|  Swarm: inactive
|  Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
|  Default Runtime: runc
|  Init Binary: docker-init
|  containerd version: 2456e983eb9e37e47538f59ea18f2043c9a73640
|  runc version: v1.1.4-0-g5fd4c4d
|  init version: de40ad0
|  Security Options:
|   seccomp
|    Profile: default
|   cgroupns
|  Kernel Version: 5.15.49-linuxkit
|  Operating System: Docker Desktop
|  OSType: linux
|  Architecture: aarch64
|  CPUs: 6
|  Total Memory: 7.667GiB
|  Name: docker-desktop
|  ID: 47K2:6RCM:6WH6:HQ22:ADYC:DHJD:LYUL:765E:ASXH:T4O3:DAAH:GQIZ
|  Docker Root Dir: /var/lib/docker
|  Debug Mode: false
|  HTTP Proxy: http.docker.internal:3128
|  HTTPS Proxy: http.docker.internal:3128
|  No Proxy: hubproxy.docker.internal
|  Registry: https://index.docker.io/v1/
|  Labels:
|  Experimental: false
|  Insecure Registries:
|   hubproxy.docker.internal:5000
|   127.0.0.0/8
|  Live Restore Enabled: false
| 
[Pipeline/test-build]   ❓  ::endgroup::
[Pipeline/test-build]   ❓  ::group::Buildx version
| [command]/usr/bin/docker buildx version
| github.com/docker/buildx 0.10.0+azure-1 876462897612d36679153c3414f7689626251501
[Pipeline/test-build]   ❓  ::endgroup::
| [command]/usr/bin/docker buildx build --iidfile /tmp/docker-build-push-jRPMOi/iidfile --tag ghcr.io/elifesciences/enhanced-preprints:test --load --metadata-file /tmp/docker-build-push-jRPMOi/metadata-file https://github.com/elifesciences/enhanced-preprints-server.git#430e401d4a8cc473198d8dd1141e6e0fe53619b7
| #1 [internal] load git source https://github.com/elifesciences/enhanced-preprints-server.git#430e401d4a8cc473198d8dd1141e6e0fe53619b7
| #1 CACHED
| 
| #2 [internal] load metadata for docker.io/library/node:16.18-alpine3.15
| #2 DONE 0.4s
| 
| #3 [base 1/3] FROM docker.io/library/node:16.18-alpine3.15@sha256:ecf74556cdeee48382e555a377ddb12d36161bd33349dc79290f733f763df711
| #3 DONE 0.0s
| 
| #4 [prod 1/3] COPY --from=build /opt/epp/node_modules node_modules
| #4 CACHED
| 
| #5 [build 6/6] RUN yarn
| #5 CACHED
| 
| #6 [build 5/6] COPY .yarn/releases .yarn/releases
| #6 CACHED
| 
| #7 [prod 2/3] COPY package.json package.json
| #7 CACHED
| 
| #8 [build 3/6] COPY yarn.lock yarn.lock
| #8 CACHED
| 
| #9 [base 2/3] RUN mkdir /opt/epp
| #9 CACHED
| 
| #10 [build 1/6] RUN apk add git python3 make gcc musl-dev g++
| #10 CACHED
| 
| #11 [base 3/3] WORKDIR /opt/epp
| #11 CACHED
| 
| #12 [build 2/6] COPY package.json package.json
| #12 CACHED
| 
| #13 [build 4/6] COPY .yarnrc.yml .yarnrc.yml
| #13 CACHED
| 
| #14 [prod 3/3] COPY src/ src/
| #14 CACHED
| 
| #15 exporting to image
| #15 exporting layers done
| #15 writing image sha256:17d88f082634cb37f52a1af486ce76d8e45c36895cd8e916cb56f3b974d957b5 done
| #15 naming to ghcr.io/elifesciences/enhanced-preprints:test done
| #15 DONE 0.0s
[Pipeline/test-build]   ❓  ::group::ImageID
| sha256:17d88f082634cb37f52a1af486ce76d8e45c36895cd8e916cb56f3b974d957b5
[Pipeline/test-build]   ❓  ::endgroup::
[Pipeline/test-build]   ❓  ::group::Digest
| sha256:17d88f082634cb37f52a1af486ce76d8e45c36895cd8e916cb56f3b974d957b5
[Pipeline/test-build]   ❓  ::endgroup::
[Pipeline/test-build]   ❓  ::group::Metadata
| {
|   "containerimage.digest": "sha256:17d88f082634cb37f52a1af486ce76d8e45c36895cd8e916cb56f3b974d957b5"
| }
[Pipeline/test-build]   ❓  ::endgroup::
[Pipeline/test-build]   ✅  Success - Main Build test
[Pipeline/test-build]   ⚙  ::set-output:: metadata={
  "containerimage.digest": "sha256:17d88f082634cb37f52a1af486ce76d8e45c36895cd8e916cb56f3b974d957b5"
}
[Pipeline/test-build]   ⚙  ::set-output:: imageid=sha256:17d88f082634cb37f52a1af486ce76d8e45c36895cd8e916cb56f3b974d957b5
[Pipeline/test-build]   ⚙  ::set-output:: digest=sha256:17d88f082634cb37f52a1af486ce76d8e45c36895cd8e916cb56f3b974d957b5
[Pipeline/test-build] ⭐ Run Main Test docker run
[Pipeline/test-build]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/2] user= workdir=
| 6bf43e3b4dc64f284e4b9b7da2fc9c31d6db10ac3dfe0e4e766925d8db097693
[Pipeline/test-build]   ✅  Success - Main Test docker run
[Pipeline/test-build] ⭐ Run Main Check status code
[Pipeline/test-build]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/3] user= workdir=
| *   Trying ::1...
| * TCP_NODELAY set
| * connect to ::1 port 3000 failed: Connection refused
| *   Trying 127.0.0.1...
| * TCP_NODELAY set
| * connect to 127.0.0.1 port 3000 failed: Connection refused
| * Failed to connect to localhost port 3000: Connection refused
| * Closing connection 0
| Warning: Transient problem: connection refused Will retry in 1 seconds. 5 
| Warning: retries left.
| * Hostname localhost was found in DNS cache
| *   Trying ::1...
| * TCP_NODELAY set
| * connect to ::1 port 3000 failed: Connection refused
| *   Trying 127.0.0.1...
| * TCP_NODELAY set
| * connect to 127.0.0.1 port 3000 failed: Connection refused
| * Failed to connect to localhost port 3000: Connection refused
| * Closing connection 1
| Warning: Transient problem: connection refused Will retry in 2 seconds. 4 
| Warning: retries left.
| * Hostname localhost was found in DNS cache
| *   Trying ::1...
| * TCP_NODELAY set
| * connect to ::1 port 3000 failed: Connection refused
| *   Trying 127.0.0.1...
| * TCP_NODELAY set
| * connect to 127.0.0.1 port 3000 failed: Connection refused
| * Failed to connect to localhost port 3000: Connection refused
| * Closing connection 2
| Warning: Transient problem: connection refused Will retry in 4 seconds. 3 
| Warning: retries left.
| * Hostname localhost was found in DNS cache
| *   Trying ::1...
| * TCP_NODELAY set
| * connect to ::1 port 3000 failed: Connection refused
| *   Trying 127.0.0.1...
| * TCP_NODELAY set
| * Connected to localhost (127.0.0.1) port 3000 (#3)
| > GET /api/reviewed-preprints HTTP/1.1
| > Host: localhost:3000
| > User-Agent: curl/7.58.0
| > Accept: */*
| > 
| < HTTP/1.1 200 OK
| < X-Powered-By: Express
| < Content-Type: application/json; charset=utf-8
| < Content-Length: 22
| < ETag: W/"16-IfdoExwkJFY0ev5dtvGKgtFEuV8"
| < Date: Fri, 03 Mar 2023 14:30:21 GMT
| < Connection: keep-alive
| < Keep-Alive: timeout=5
| < 
| * Connection #3 to host localhost left intact
[Pipeline/test-build]   ✅  Success - Main Check status code
[Pipeline/test-build] ⭐ Run Post Build test
[Pipeline/test-build]   🐳  docker exec cmd=[node /var/run/act/actions/docker-build-push-action@v4/dist/index.js] user= workdir=
[Pipeline/test-build]   ❓  ::group::Removing temp folder /tmp/docker-build-push-jRPMOi
[Pipeline/test-build]   ❓  ::endgroup::
[Pipeline/test-build]   ✅  Success - Post Build test
[Pipeline/test-build] 🏁  Job succeeded
```